### ISSUE TITLE
Ensure proper numerics for RetinaNet

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#1308]
I'm not sure if the RetinaNet still gets good scores using the training script.  At some point, something broke.  Not sure what yet.

**Comment by LukeWood** [comment]
Two PRs to check:
- [dictbox PR]
- [retinanet loss reduction logic update](https://github.com/keras-team/keras-cv/pull/1204/)

**Comment by LukeWood** [comment]
https://github.com/LukeWood/OD-Benchmarks
